### PR TITLE
Create scripts for every test target's configuration

### LINKED ISSUE #7012: Resources are not installed for all test target's configurations except Debug
# Report

## What did you do?

Create a test specification in pod's podspec file containing `resource_bundles`.
Specify test spec with `:testspecs => ['test_spec_name']` for the corresponding pod in Podfile.
Run `pod install`
Edit test target's scheme to run tests for any configuration other than **Debug** (e.g. **Release**).
Run tests for test target.

## What did you expect to happen?

Resources from test specification are installed (copied into test bundle). Tests relying on the resources pass.

## What happened instead?

Resources are not installed (no resource bundle in test bundle). Tests relying on the resources fail.

## CocoaPods Environment

```
   CocoaPods : 1.3.1
        Ruby : ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin16]
    RubyGems : 2.5.1
        Host : Mac OS X 10.12.6 (16G29)
       Xcode : 8.3.3 (8E3004b)
         Git : git version 2.11.0
Ruby lib dir : /Users/nick/.rbenv/versions/2.3.1/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ d5a73389c142c2ba70e7d659e1cfce54fa9701d1
```

### Installation Source

```
Executable Path: /Users/nick/.rbenv/versions/2.3.1/bin/pod
```

### Plugins

```
claide-plugins        : 0.9.2
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.2.0
cocoapods-try         : 1.1.0
slather               : 2.4.1
```

### Other notes 

The problem seems to be coming from here
https://github.com/CocoaPods/CocoaPods/blob/19203fc02cc0ce57ef221a87aa958ea0a04d7770/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb#L342

Resource paths map is hardcoded with 'Debug' configuration only ignoring target's configurations completely.

### PR DESCRIPTION
**Pull request by tarbayev**
_Copy resource_ and _embed frameworks_ scripts of a test target now contain all the target's configurations instead of just the **Debug** one. 

This resolves #7012

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Create scripts for every test target's configuration  
  [tarbayev](https://github.com/tarbayev)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by dnkoutso** [comment]
Need a change log entry

**Comment by dnkoutso** [comment]
And run `rubocop -a`

File: spec/unit/installer/xcode/pods_project_generator/pod_target_installer_spec.rb [hunk 1]
```diff
@@ -277,6 +277,36 @@ class PodsProjectGenerator
                 end
               end
 
+              it 'creates embed frameworks script for test target' do
+                @coconut_pod_target.stubs(:requires_frameworks? => true)
+                @installer.install!
+
```
Comment (dnkoutso) [inline spec/unit/installer/xcode/pods_project_generator/pod_target_installer_spec.rb]: remove empty lines here.

**Comment by dnkoutso** [comment]
I think this is not 100% done yet. `pod_target.rb` method `framework_paths` returns the full flat list of all paths for all configurations.

This might need to change to actually return a hash of arrays where each key in the hash is the configuration name and the value is the the list of framework paths.

The same should apply for `resource_paths` method in `pod_target.rb`. 

`aggregate_target.rb` method `framework_paths_by_config` does it correctly.

**Comment by dnkoutso** [comment]
Hm I might be wrong and need to think about this more. I think `pod_target` does not differentiate between the different build configurations since we never create `PodA-Debug` and `PodA-Release` variants.

Your change might be enough let me think about it a bit more.

**Comment by tarbayev** [comment]
@dnkoutso I was looking at `aggregate_target.rb` as a reference, but found the logic there as non-relative and decided to just add missing configurations.

**Comment by dnkoutso** [comment]
needs a rebase but I think looks good!

File: spec/fixtures/coconut-lib/Tests/CoconutTest.xib [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (dnkoutso) [inline spec/fixtures/coconut-lib/Tests/CoconutTest.xib]: is this file needed in order for the tests to pass?

Comment (tarbayev) [reply spec/fixtures/coconut-lib/Tests/CoconutTest.xib]: seems like it's not actually

**Comment by tarbayev** [comment]
Looks like some random failure, not related to my changes.

**Comment by dnkoutso** [comment]
thanks!
